src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -2509,6 +2509,52 @@ bool AdaptDispatch::DoITerm2Action(const std::wstring_view string)
     return false;
 }
 
+// Method Description:
+// - Performs a FinalTerm action
+// - Ascribes to the ITermDispatch interface
+// - Currently, the actions we support are:
+//   * `OSC133;A`: mark a line as a prompt line
+// - Not actually used in conhost
+// - The remainder of the FTCS prompt sequences are tracked in GH#11000
+// Arguments:
+// - string: contains the parameters that define which action we do
+// Return Value:
+// - false in conhost, true for the SetMark action, otherwise false.
+bool AdaptDispatch::DoFinalTermAction(const std::wstring_view string)
+{
+    // This is not implemented in conhost.
+    if (_api.IsConsolePty())
+    {
+        // Flush the frame manually, to make sure marks end up on the right line, like the alt buffer sequence.
+        _renderer.TriggerFlush(false);
+        return false;
+    }
+
+    const auto parts = Utils::SplitString(string, L';');
+
+    if (parts.size() < 1)
+    {
+        return false;
+    }
+
+    const auto action{ parts[0] };
```

Comment:
I don't believe this code requires a string copy. This avoids one:
```suggestion
    const auto& action = parts[0];
```

Reply:
> I see you prefer he not use the uniform initialization syntax (since you switched it to `=`) 😉

Yeah, the rest of the file uses classic `=` assignments. I like consistency. 👉👈
Of course you probably already figured out my deepest secret that - entirely unrelated to this - I personally kinda dislike the uniform initialization syntax. 😅 This is what I write:
```cpp
auto bar = bar_producer();  // good
auto bar{ bar_producer() }; // "bad"
Foo foo{ bar };             // good
```
Generally speaking, I dislike the uniform initialization syntax. Isn't it just a bolted on language hack, because people were afraid of breaking existing code? I use it because it's useful, but would you write this if it was an equivalent thing?
```cpp
auto bar(bar_producer());
```
I wouldn't and so I don't use curly brackets either, because I consider curly brackets stricter round brackets. In my mind I'm not constructing an object there, I'm assigning an unnamed rvalue a name via variable assignment. And so I write `auto result = foo()`.

> _I had to read the rules about `const auto&` locals and how they extend lifetime -- I still think it's a bad pattern to encourage, but we use it all over the place \\\_(shrug)\_\/_

I think about it differently: Here we "bind to the memory address" of the first item in an array.

Reply:
ultimately, Audit will complain:

```
error C26445: Do not assign gsl::span or std::string_view to a reference. They are cheap to construct and are not owners of the underlying data.
```

so, `const auto action = ...` it is

Reply:
> error C26445: Do not assign gsl::span or std::string_view to a reference.

But passing it as a by-reference argument is fine? smh